backend/src/main.cpp

```diff
@@ -9,29 +9,68 @@
 #include "server.hpp"
 #include "sodium.h"
 #include "cryptography.hpp"
+#include "spdlog/spdlog.h"
 
-int main()
+int main(int argc, char *argv[])
 {
-    database::Database db = database::Database("passwords.db");
-    db.execute("CREATE TABLE passwords (password TEXT);");
-
-    // generate and insert the passwords into the database
-    std::unordered_set<std::string> passwords = password::generatePasswords(100, 20);
-    passwords.insert("TestPass1&");
-    passwords.insert("ChocolateCake1!");
-    // 1. generate secret key b
-    unsigned char b[crypto_core_ristretto255_SCALARBYTES];
-    crypto_core_ristretto255_scalar_random(b);
-    // 2. encrypt each password with b (and hash to point)
-    std::vector<std::string> encrypted_passwords = cryptography::encrypt(passwords, b);
-
-    // 3. insert into database
-    for (const auto &password : passwords)
+    spdlog::set_level(spdlog::level::debug);
+
+    if (argc < 2 || argc > 3)
+    {
+        printf("Usage: %s <database file> [--build]\n", argv[0]);
+        return 1;
+    }
+
+    bool build = false;
+    if (argc == 3)
+    {
+        if (std::string(argv[2]) == "--build")
+        {
+            build = true;
+        }
+        else
+        {
+            printf("Usage: %s <database file> [--build]\n", argv[0]);
+            return 1;
+        }
+    }
+
+    database::Database db = database::Database(argv[1], build);
```

Comment:
This is hard to test in `test/database.cpp` since it is in `main.cpp` but I manually tested it. When we write integration tests we can go further and test it throws an invalid argument error.